doc/development/plugins.rst

```diff
@@ -105,22 +114,22 @@ then finite shots exist. If ``shots`` is falsy, then an analytic execution shoul
 Preprocessing
 -------------
 
-The preprocessing method has two main responsibilities:
+The :meth:`~.devices.Device.preprocess` method has two main responsibilities:
 
 1) Create a :class:`~.TransformProgram` capable of turning an arbitrary batch of :class:`~.QuantumScript`\ s into a new batch of tapes supported by the ``execute`` method.
 2) Setup the :class:`~.ExecutionConfig` dataclass by filling in device options and making decisions about differentiation.
 
-Once the transform program has been applied to a batch of circuits, that batch should be able to run via ``Device.execute`` without error.
+These two tasks can be extracted into private methods or helper functions if that improves source
```

Comment:
We should probably make a mention here that that preprocessing be created by manually constructing the `preprocess` method, providing a toml file, or creating a default version from the toml file before extending and customizing the default behavior.

Developers shouldn't learn this information after they've already learned all about setting up a custom preprocessing.

Reply:
Done